frontend/src/views/admin/assignments/accept-offer-button-with-dialog.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { ActionButton } from "../../../components/action-buttons";
+import { OfferConfirmationDialog } from "./offer-confirmation-dialog";
+import { Assignment } from "../../../api/defs/types";
+import { FaCheck } from "react-icons/fa";
+
+export function AcceptOfferButtonWithDialog(props: {
```

Comment:
Probably all in in one file! Because to destructure props in function arguments would require defining a interface to recognize props type, which should be re-used